src/nvim/ops.c

```diff
@@ -516,8 +515,9 @@ static void block_insert(oparg_T *oap, char_u *s, int b_insert, struct block_def
     memset(newp + offset, ' ', (size_t)spaces);
 
     /* copy the new text */
-    memmove(newp + offset + spaces, s, (size_t)s_len);
-    offset += s_len;
+    memmove(newp + offset + spaces, s, s_len);
+    assert(s_len <= INT_MAX);
```

Comment:
assert can be removed, as inserted text `s` (inserted by the user on the first line of the block) couldn't be longer than MAXCOL to start with.